### PR TITLE
document describing a simple usage of `coalton-testing`

### PR DESCRIPTION
**Pull request by gefjon**
Note that there are a handful of TODO lines in there.

### CODE REVIEW

File: docs/intro-to-coalton-testing.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+the project's root directory, which looks something like:
+
+```
+(defsystem "foo"
+  :depends-on ("coalton" "named-readtables" ...other dependencies?)
+  :components ((:file "main")
+               ...other components?))
+```
+
+Don't worry if your system looks a little different. It might have `:serial t`, and it
```
Comment (eliaslfox) [inline docs/intro-to-coalton-testing.md:22]: I think this note is a little superfluous.

File: docs/intro-to-coalton-testing.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+(defsystem "foo/test"
+  :depends-on ("foo" "coalton/testing")
+  :pathname "test/"
+  :serial t
+  :components ((:file "test"))
+  :perform (test-op (o s)
+             (symbol-call '#:foo/test '#:run-tests)))
+```
+
+TODO: do we want to tell users to explicitly depend on `fiasco`, the same way we tell them
```
Comment (eliaslfox) [inline docs/intro-to-coalton-testing.md:50]: I guess so

File: docs/intro-to-coalton-testing.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+`define-test` is a macro that was generated by the `coalton-fiasco-init` form you wrote
+earlier. (This is an implementation detail; try not to think too much about it.) For your
+very first `define-test`, try writing:
+
+```
+(define-test my-first-test ()
+  (is True))
+```
+
+`my-first-test` is the name of your test. The empty list `()` is its argument list. I
```
Comment (eliaslfox) [inline docs/intro-to-coalton-testing.md:138]: Probably an unnecessary note for a tutorial.

File: docs/intro-to-coalton-testing.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+(define-test test-one-element-list ()
+  (match (one-element-list 0)
+    ((Nil) (is False "ONE-ELEMENT-LIST returned an empty list!"))
+    ((Cons elt (Nil)) (is (== 0 elt)))
+    ((Cons _ _) (is False "ONE-ELEMENT-LIST returned a list with more than 1 element!"))))
+```
+
+## Running your tests
+
+Now that you've defined a test system and some tests to go in it, you can run your tests
```
Comment (eliaslfox) [inline docs/intro-to-coalton-testing.md:219]: There should be a note here that running asdf commands should not be done in a coalton package. Doing so tends to corrupt asdf.

**Review by eliaslfox — COMMENTED** [review]
Thanks for taking the time to document this. I think it would be worth also having an example project skeleton, something like [coalton-example](https://github.com/eliaslfox/coalton-example).

File: docs/intro-to-coalton-testing.md (hunk 4 above)
Comment (gefjon) [reply docs/intro-to-coalton-testing.md:219]: Is this still the case now that `coalton` reexports `defpackage` and `in-package`? And if so, is it a breakage between Coalton and ASDF, or is it that some other systems expect to be in a package that uses CL before being loaded, and contain code with that expectation before defining and entering their own packages?

**Comment by gefjon** [comment]
Still one TODO: Fiasco and FiveAM's `is` macros both privilege the argument order `(is (PREDICATE EXPECTED-VALUE COMPUTED-VALUE))` when writing `is` tests of binary comparisons, generating error messages that describe the first argument as the expected value, and the second argument as the value computed by the tests. `coalton/testing` does not do this. Currently, the intro-to-coalton-testing recommends that users obey that convention anyways. Do we want to document preferring this convention? Personally, I lean towards yes (I mean, I wrote it that way, duh) because I find it easy to read, and I value consistency with the rest of the CL world, but I can see an argument for removing the recommendation from the document, or adding language that specifies to prefer it because it's a convenient convention, rather than just saying to prefer it.

**Comment by eliaslfox** [comment]
I always found the expected/actual designations to be a bit arbitrary. I also frequently get it "wrong" in my own code. The [fiasco readme](https://github.com/joaotavora/fiasco) seems to use both styles.

I personally wouldn't recommend a specific style, but I think the wording in the document is fine as is.

**Comment by eliaslfox** [comment]
Sorry, I completely forgot about this.

**Review by eliaslfox — APPROVED** [review]

**Pull request merged**